tests/runtime/filter_aws.c

```diff
@@ -0,0 +1,269 @@
+    struct flb_aws_client_generator *client_generator;
+    struct flb_filter_aws_init_options *ops;
+    flb_sds_t name;
+    struct flb_aws_client_mock_request_chain *request_chain;
+    char *output = NULL;
+
+    request_chain = FLB_AWS_CLIENT_MOCK(
+        response(
+            expect(URI, "/latest/meta-data/tags/instance"),
+            expect(METHOD, FLB_HTTP_GET),
+            expect(HEADER_COUNT, 1),
+            set(STATUS, 404),
+        ),
+    );
+    flb_aws_client_mock_configure_generator(request_chain);
+
+    client_generator = flb_aws_client_get_mock_generator();
+    ops = flb_calloc(1, sizeof(struct flb_filter_aws_init_options));
+    name = flb_sds_create_size(100);
+    name = flb_sds_printf(&name, "flb_test_aws_ec2_tags_404\0");
+    ops->client_generator = client_generator;
+    ops->name = name;
+
+    ctx = flb_create();
+
+    in_ffd = flb_input(ctx, (char *) "lib", NULL);
+    TEST_CHECK(in_ffd >= 0);
+    flb_input_set(ctx, in_ffd, "tag", "test", NULL);
+
+
+    /* Prepare output callback context*/
+    cb_data.cb = callback_test;
+    cb_data.data = NULL;
+
+    /* Lib output */
+    out_ffd = flb_output(ctx, (char *) "lib", (void *)&cb_data);
+    TEST_CHECK(out_ffd >= 0);
+    flb_output_set(ctx, out_ffd,
+                   "match", "*",
+                   "format", "json",
+                   NULL);
+
+    filter_ffd = flb_filter(ctx, (char *) "aws", ops);
+    TEST_CHECK(filter_ffd >= 0);
+    ret = flb_filter_set(ctx, filter_ffd, "match", "*", NULL);
+    TEST_CHECK(ret == 0);
+    ret = flb_filter_set(ctx, filter_ffd, "ec2_instance_id", "false", NULL);
+    TEST_CHECK(ret == 0);
+    ret = flb_filter_set(ctx, filter_ffd, "az", "false", NULL);
+    TEST_CHECK(ret == 0);
+    ret = flb_filter_set(ctx, filter_ffd, "tags_enabled", "true", NULL);
+    TEST_CHECK(ret == 0);
+
+    ret = flb_start(ctx);
+    TEST_CHECK(ret == 0);
+
+    bytes = flb_lib_push(ctx, in_ffd, p, strlen(p));
+    if (!TEST_CHECK(bytes > 0)) {
+        TEST_MSG("zero bytes were pushed\n");
+    }
+
+    flb_time_msleep(1000);
+
+    output = get_output();
+    if (output) {
+        char *result = strstr(output, "\"Name\":\"my_ec2_instance\"");
+        if (!TEST_CHECK(result == NULL)) {
+            TEST_MSG("output:%s\n", output);
+        }
+        result = strstr(output, "\"CUSTOMER_ID\":\"70ec5c04-3a6e-11ed-a261-0242ac120002\"");
+        if (!TEST_CHECK(result == NULL)) {
+            TEST_MSG("output:%s\n", output);
+        }
+        result = strstr(output, "hello, from my ec2 instance");
+        if (!TEST_CHECK(result != NULL)) {
+            TEST_MSG("output:%s\n", output);
+        }
+    }
+    else {
+        TEST_CHECK(false);
+        TEST_MSG("output is empty");
+    }
+
+    flb_stop(ctx);
+    flb_destroy(ctx);
+    flb_free(ops);
+
+    set_output(NULL);
+    // TODO: Fix destroying the aws client mock generator.
+    //     It is commented out to avoid segvs.
+    // flb_aws_client_mock_destroy_generator();
```

Comment:
Here is the commented out `flb_aws_client_mock_destroy_generator` which causes a SEGV. Let me know if you have any tips how to destroy the generator.